zerver/tests/test_subs.py

```diff
@@ -374,48 +375,86 @@ class StreamAdminTest(ZulipTestCase):
         params = {
-            'stream_name': ujson.dumps('private_stream'),
+            'stream_name': ujson.dumps('private_stream_1'),
             'is_private': ujson.dumps(False),
         }
         result = self.client_patch(f"/json/streams/{stream_id}", params)
         self.assert_json_success(result)
 
         realm = user_profile.realm
-        stream = get_stream('private_stream', realm)
+        stream = get_stream('private_stream_1', realm)
+        self.assertFalse(stream.invite_only)
+        self.assertTrue(stream.history_public_to_subscribers)
+
+        do_change_user_role(user_profile, UserProfile.ROLE_MEMBER)
+        params = {
+            'stream_name': ujson.dumps('private_stream_2'),
+            'is_private': ujson.dumps(False),
+        }
+        stream = self.subscribe(user_profile, 'private_stream_2')
+        result = self.client_patch(f"/json/streams/{stream.id}", params)
+        self.assert_json_error(result, "This action requires administrative rights.")
+
+        sub = get_subscription('private_stream_2', user_profile)
+        do_change_subscription_property(user_profile, sub, stream, "role", Subscription.ROLE_STREAM_ADMINISTRATOR)
+        result = self.client_patch(f"/json/streams/{stream.id}", params)
+        self.assert_json_success(result)
+
+        stream = get_stream('private_stream_2', realm)
         self.assertFalse(stream.invite_only)
         self.assertTrue(stream.history_public_to_subscribers)
 
     def test_make_stream_private(self) -> None:
         user_profile = self.example_user('hamlet')
         self.login_user(user_profile)
         realm = user_profile.realm
-        self.make_stream('public_stream', realm=realm)
+        self.make_stream('public_stream_1', realm=realm)
+        self.make_stream('public_stream_2')
 
         do_change_user_role(user_profile, UserProfile.ROLE_REALM_ADMINISTRATOR)
         params = {
-            'stream_name': ujson.dumps('public_stream'),
+            'stream_name': ujson.dumps('public_stream_1'),
             'is_private': ujson.dumps(True),
         }
-        stream_id = get_stream('public_stream', realm).id
+        stream_id = get_stream('public_stream_1', realm).id
         result = self.client_patch(f"/json/streams/{stream_id}", params)
         self.assert_json_success(result)
-        stream = get_stream('public_stream', realm)
+        stream = get_stream('public_stream_1', realm)
+        self.assertTrue(stream.invite_only)
+        self.assertFalse(stream.history_public_to_subscribers)
+
+        do_change_user_role(user_profile, UserProfile.ROLE_MEMBER)
+        params = {
+            'stream_name': ujson.dumps('public_stream_2'),
+            'is_private': ujson.dumps(True),
+        }
+        stream = self.subscribe(user_profile, 'public_stream_2')
+        result = self.client_patch(f"/json/streams/{stream.id}", params)
+        self.assert_json_error(result, "This action requires administrative rights.")
```

Comment:
Let's add a chekc here that `stream.invite_only` is False (Etc.) -- that the change didn't happen in addition to it throwing the error.  Same with the last test.